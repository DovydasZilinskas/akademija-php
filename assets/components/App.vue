<template>
  <div>
    <h2 class="center">Contact Me</h2>
  </div>
</template>

<script>
export default {
  name: 'app',
};
</script>

<style scoped>
.wrapper {
  margin: 0 auto;
}
</style>
