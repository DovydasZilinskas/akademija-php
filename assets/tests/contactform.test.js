import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import App from '../views/ContactForm.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('App', () => {
  // let actions
  let getters;
  let store;
  let mutations;

  beforeEach(() => {
    getters = {
      getCurrentName: () => 'Name',
      getCurrentEmail: () => 'Email',
      getCurrentMessage: () => 'Message',
    };
    mutations = {
      updateName: () => 'updateName',
    };
    store = new Vuex.Store({
      getters,
      mutations,
    });
  });

  it('expected?', () => {
    const wrapper = shallowMount(App, { store, localVue });
    expect(wrapper).toMatchSnapshot();
  });

  it('find name field', async () => {
    const wrapper = shallowMount(App, { store, localVue });
    const input = wrapper.find('input[type="text"]');
    await input.setValue('Name');

    const state = {
      name: 'Name',
    };

    expect(wrapper.find('input[type="text"]').element.value).toBe(
      getters.getCurrentName(state)
    );
  });
});
