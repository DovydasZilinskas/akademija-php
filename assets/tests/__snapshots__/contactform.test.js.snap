// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App expected? 1`] = `
<div class="container">
  <h2>Contact me</h2>
  <form><label>Full Name</label> <input type="text" maxlength="255" placeholder="Name"> <label>Email</label> <input type="email" maxlength="255" placeholder="Email"> <label>Message</label> <textarea maxlength="255" placeholder="Your text here..."></textarea>
    <vue-recaptcha-stub sitekey="6Ld2MlUaAAAAABKG7ibjB5iZIXVDLBy_e5sSVLJq" size="invisible" recaptchascriptid="__RECAPTCHA_SCRIPT" recaptchahost="www.google.com" language="" class="g-recaptcha"></vue-recaptcha-stub> <button type="submit">Send</button>
  </form>
  <!---->
  <spinner-stub></spinner-stub>
</div>
`;
